index.js

```diff
@@ -109,7 +110,11 @@ var Node = function (path, opts, disposable) {
             var match = (data + '').trim().match(/API server listening on (.*)/)
             if (match) {
               t.apiAddr = match[1]
-              cb(null, ipfs(t.apiAddr))
+              var addr = multiaddr(t.apiAddr).nodeAddress()
+              var api = ipfs(t.apiAddr)
+              api.host = addr.address
+              api.port = addr.port
```

Comment:
this should probably be `api.apiAddr = t.apiAddr` because:
- want to stick to multiaddrs. this might not always be HTTP/TCP
- want to be explicit it's the addr of the api, because:

```
ipfs.host
ipfs.addr
```
might imply the node's actual (api protocol) address, instead of the api addr.